src/error.rs

```diff
@@ -79,27 +89,38 @@ impl Error {
     #[inline]
     pub fn raw_os_error(self) -> Option<RawOsError> {
         let code = self.0.get();
-        if code >= Self::INTERNAL_START {
-            return None;
+        #[cfg(target_os = "uefi")]
+        {
+            if code & UEFI_ERROR_FLAG != 0 {
+                Some(code)
+            } else {
+                None
+            }
+        }
+
+        #[cfg(not(target_os = "uefi"))]
+        {
+            if code >= 0 {
+                return None;
+            }
+            #[cfg(not(target_os = "solid_asp3"))]
+            let code = code.checked_neg()?;
```

Comment:
I am not sure whether we should do this negation on WASIp1 and Hermit.

Reply:
I think we should figure out where they define their error constants, and use whatever convention is in place there.

Reply:
We probably should look into what is done by `std`. After all, the primary use case for the error codes is to get error strings which are included with `std`/`libc`.

Reply:
I added the comment for all targets which rely on `from_os_error`: https://github.com/rust-random/getrandom/blob/error_tweak/src/error.rs#L106-L114